km61/Kovtun_Artem/laboratory1.sql

```diff
@@ -1,2 +1,137 @@
+DEFAULT TABLESPACE 'USERS'
+TEMPORARY TABLESPACE 'TEMP';
+
+ALTER USER kovtun QUOTA 10M;
+
+GRANT SELECT ANY TABLE TO kovtun;
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Громадянин України має власне житло та автомобіль.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE TABLE humans 
+(
+    human_id NUMBER(6) NOT NULL,
+    human_name VARCHAR2(10) NOT NULL
+);
+
+ALTER TABLE humans ADD CONSTRAINT human_id_pk PRIMARY KEY (human_id);
+
+CREATE TABLE appartments 
+(
+    app_id NUMBER(6) NOT NULL,
+    app_quantity_of_rooms NUMBER(2) NOT NULL
+);
+
+ALTER TABLE appartments ADD CONSTRAINT app_id_pk PRIMARY KEY (app_id);
+
+CREATE TABLE cars
+(
+    car_id NUMBER(6) NOT NULL,
+    car_tupe VARCHAR(20) NOT NULL
+);
+
+ALTER TABLE cars ADD CONSTRAINT car_id_pk PRIMARY KEY (car_id);
+
+
+CREATE TABLE human_appartment 
+(
+    human_id_fk NUMBER(6) NOT NULL,
+    app_id_fk NUMBER(6) NOT NULL
+);
+
+ALTER TABLE human_appartment ADD CONSTRAINT human_appartment_pk PRIMARY KEY (human_id_fk, app_id_fk);
+
+ALTER TABLE human_appartment ADD CONSTRAINT human_fk FOREIGN KEY (human_id_fk) REFERENCES human(human_id);
+
+ALTER TABLE human_appartment ADD CONSTRAINT app_fk FOREIGN KEY (app_id_fk) REFERENCES appartments(app_id);
+
+
+
+CREATE TABLE human_car 
+(
+    human_id_fk NUMBER(6) NOT NULL,
+    car_id_fk NUMBER(6) NOT NULL
+);
+
+ALTER TABLE human_car ADD CONSTRAINT human_car_pk PRIMARY KEY (human_id_fk, app_id_fk);
+
+ALTER TABLE human_car  ADD CONSTRAINT human_fk FOREIGN KEY (human_id_fk) REFERENCES human(human_id);
+
+ALTER TABLE human_car ADD CONSTRAINT app_fk FOREIGN KEY (car_id_fk) REFERENCES appartments(app_id);
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+GRANT CREATE ANY TABLE TO kovtun;
+GRANT UPDATE ANY TABLE TO kovtun;
+GRANT SELECT ANY TABLE TO kovtun;
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть постачальника, що продав найдешевший товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+PROJECT VENDORS TIMES PRODUCTS TIMES ORDERITEMS WHERE 
+    VENDORS.VEND_ID = PRODUCTS.VEND_ID AND PRODUCTS.PROD_ID IN (PROJECT ORDERSITEMS (MIN()) {VENDOR.VEND_NAME} 
+
+
+/*---------------------------------------------------------------------------
```

Comment:
SELECT VENDOR.VEND_NAME 
      FROM VENDORS, PRODUCTS, ORDERITEMS 
WHERE 
    VENDORS.VEND_ID = PRODUCTS.VEND_ID AND PRODUCTS.PROD_ID IN (SELECT MIN(product.item_price)
FROM ORDERITEMS  )